### PR TITLE
Rotation api extension

### PR DESCRIPTION
**Pull request by RobWalt**
# Objective

- Another way of specifying rotations was requested in https://github.com/bevyengine/bevy/issues/11132#issuecomment-2344603178

## Solution

- Add methods on `Rot2`
  - `turn_fraction(fraction: f32) -> Self`
  - `as_turn_fraction(self) -> f32`
- Also add some documentation on range of rotation

## Testing

- extended existing tests
- added new tests

## Showcase 

```rust
let rotation1 = Rot2::degrees(90.0);
let rotation2 = Rot2::turn_fraction(0.25);

// rotations should be equal
assert_relative_eq!(rotation1, rotation2);

// The rotation should be 90 degrees
assert_relative_eq!(rotation2.as_radians(), FRAC_PI_2);
assert_relative_eq!(rotation2.as_degrees(), 90.0);

```

### CODE REVIEW

File: crates/bevy_math/src/rotation2d.rs [hunk 1]
```diff
@@ -100,18 +102,36 @@ impl Rot2 {
     };
 
     /// Creates a [`Rot2`] from a counterclockwise angle in radians.
+    ///
+    /// # Note
+    ///
+    /// The resulting rotation will always be clamped to the range (-PI, PI] by design
```
Comment (Jondolf) [inline crates/bevy_math/src/rotation2d.rs:108]: This and the other similar comments should end in a period.

```suggestion
    /// The resulting rotation will always be clamped to the range (-PI, PI] by design.
```

Comment (Jondolf) [reply crates/bevy_math/src/rotation2d.rs:108]: Also I noticed `as_radians` uses lowercase pi. Either that should be uppercase, or this should also be lowercase.

File: crates/bevy_math/src/rotation2d.rs [hunk 2]
```diff
@@ -100,18 +102,36 @@ impl Rot2 {
...
         Self::radians(degrees.to_radians())
     }
 
+    /// Creates a [`Rot2`] from a counterclockwise fraction of a full turn of 360 degrees.
+    ///
+    /// # Note
+    ///
+    /// The resulting rotation will always be clamped to the range (-0.5, 0.5] by design
+    #[inline]
+    pub fn turn_fraction(fraction: f32) -> Self {
```
Comment (Jondolf) [inline crates/bevy_math/src/rotation2d.rs:131]: Non-blocking: A code example might be nice to illustrate this better, something similar to the showcase in the PR description.

**Review by Jondolf — APPROVED** [review]
LGTM! I think this is a useful API. Left some non-blocking suggestions

**Comment by Jondolf** [comment]
CC @johannesvollmer; this doesn't fully address what you described in https://github.com/bevyengine/bevy/issues/11132#issuecomment-2344603178, because `Rot2` is a unit complex number that should always be normalized, so it can't represent turns greater than a full rotation, and the internal representation still isn't turns. But at the API level, I think this is still useful for rotations.

File: crates/bevy_math/src/rotation2d.rs (hunk 1 above)
Comment (Jondolf) [reply crates/bevy_math/src/rotation2d.rs:108]: Oh and also these ranges should probably be formatted as code with backticks for consistency.

**Comment by johannesvollmer** [comment]
Yes, still useful :) nice

File: crates/bevy_math/src/rotation2d.rs [hunk 3]
```diff
@@ -100,18 +102,82 @@ impl Rot2 {
     };
 
     /// Creates a [`Rot2`] from a counterclockwise angle in radians.
+    ///
+    /// # Note
+    ///
+    /// The resulting rotation will always be clamped to the range `(-pi, pi]` by design.
```
Comment (janhohenheim) [inline crates/bevy_math/src/rotation2d.rs:108]:
```suggestion
    /// The resulting rotation will always be clamped to the range `(-π, π]` by design.
```

File: crates/bevy_math/src/rotation2d.rs [hunk 4]
```diff
@@ -100,18 +102,82 @@ impl Rot2 {
...
     pub fn radians(radians: f32) -> Self {
         let (sin, cos) = ops::sin_cos(radians);
         Self::from_sin_cos(sin, cos)
     }
 
     /// Creates a [`Rot2`] from a counterclockwise angle in degrees.
+    ///
+    /// # Note
+    ///
+    /// The resulting rotation will always be clamped to the range `(-180.0, 180.0]` by design.
```
Comment (janhohenheim) [inline crates/bevy_math/src/rotation2d.rs:135]:
```suggestion
    /// The resulting rotation will always be clamped to the range `(-180.0°, 180.0°]` by design.
```

File: crates/bevy_math/src/rotation2d.rs [hunk 5]
```diff
@@ -466,12 +544,21 @@ mod tests {
         assert_relative_eq!(rotation * Dir2::Y, Dir2::NEG_X);
     }
 
+    #[test]
+    fn rotation_range() {
+        // the rotation range is `(-180, 180]` and the constructors normalize the rotations to that
+        // range
```
Comment (janhohenheim) [inline crates/bevy_math/src/rotation2d.rs:550]:
```suggestion
        // the rotation range is `(-180, 180]` and the constructors 
        // normalize the rotations to that range
```

**Review by janhohenheim — APPROVED** [review]
Nice! Left some minor documentation improvements :)
IMO the "resulting rotation" in the docs would be clearer as the "input rotation", since otherwise it reads as if the resulting rotation was once in radians, once in degrees, and once in fractions, which is not the case. But I'll leave that up to you.

File: crates/bevy_math/src/rotation2d.rs [hunk 6]
```diff
@@ -100,18 +102,82 @@ impl Rot2 {
...
     #[inline]
     pub fn degrees(degrees: f32) -> Self {
         Self::radians(degrees.to_radians())
     }
 
+    /// Creates a [`Rot2`] from a counterclockwise fraction of a full turn of 360 degrees.
+    ///
+    /// # Note
+    ///
+    /// The resulting rotation will always be clamped to the range `(-0.5, 0.5]` by design.
```
Comment (janhohenheim) [inline crates/bevy_math/src/rotation2d.rs:160]:
```suggestion
    /// The resulting rotation will always be clamped to the range `(-50%, 50%]` by design.
```
0.5 is a reasonable value in radians, so this is potentially confusing.

**Pull request merged**